pkg/controllers/disruption/helpers.go

```diff
@@ -65,20 +65,33 @@ func SimulateScheduling(ctx context.Context, kubeClient client.Client, cluster *
 		return scheduling.Results{}, errCandidateDeleting
 	}
 
-	// We get the pods that are on nodes that are deleting
-	deletingNodePods, err := deletingNodes.ReschedulablePods(ctx, kubeClient)
-	if err != nil {
-		return scheduling.Results{}, fmt.Errorf("failed to get pods from deleting nodes, %w", err)
-	}
 	// start by getting all pending pods
 	pods, err := provisioner.GetPendingPods(ctx)
 	if err != nil {
 		return scheduling.Results{}, fmt.Errorf("determining pending pods, %w", err)
 	}
+
+	// Don't provision capacity for pods which will not get evicted due to fully blocking PDBs.
+	// Since Karpenter doesn't know when these pods will be successfully evicted, spinning up capacity until
+	// these pods are evicted is wasteful.
+	pdbs, err := pdb.NewLimits(ctx, kubeClient)
+	if err != nil {
+		return scheduling.Results{}, fmt.Errorf("tracking PodDisruptionBudgets, %w", err)
+	}
 	for _, n := range candidates {
-		pods = append(pods, n.reschedulablePods...)
+		currentlyReschedulablePods := lo.Filter(n.reschedulablePods, func(p *corev1.Pod, _ int) bool {
```

Comment:
Is there a way to store this in NewCandidate so that we only store the pods that we have to model for scheduling and we somehow do the emptiness check through some other struct member? If emptiness is the only place we don't use `CurrentlReschedulable`, I question if there's a simpler way to store this

Reply:
Emptiness isn't the only place this is checked today. We also check this for things like determining if the disruption budgets are breached.

I could have a separate `currentlyReschedulablePods` struct member in `candidate` but seemed like duplicate information since it can be deducted from `reschedulablePods` although one could argue that the additional computation isn't worth it. Primarily based it off the fact that this is only fetched from `candidate` in a single place at the moment which is the above one.